### PR TITLE
[270E][next] Update dependencies 

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftersoftware/craftercms/issues/270

### CODE REVIEW

**Comment by jvega190** [comment]
@rart I need to discuss about material-ui/pickers lib that is deprecated and causing warnings

**Comment by jvega190** [comment]
As we discussed, material-ui/pickers can't be updated yet so this is a pending update to work on

File: ui/app/src/components/CommitResolutionDialog/CommitResolutionDialog.tsx [hunk 1]
```diff
@@ -19,20 +19,19 @@ import Dialog from '@material-ui/core/Dialog';
...
+import { useOnClose } from '../../utils/hooks/useOnClose';
 
 export default function CommitResolutionDialog(props: CommitResolutionDialogContainerProps) {
   const { open, onClose } = props;
   const [disableQuickDismiss, setDisableQuickDismiss] = useState(false);
+  const onCloseHandler = useOnClose();
+
+  const handleClose = (event, reason) => {
+    onCloseHandler(onClose, reason, disableQuickDismiss, disableQuickDismiss);
+  };
```
Comment (rart) [inline ui/app/src/components/CommitResolutionDialog/CommitResolutionDialog.tsx]: The idea was that the hook returns the handler directly; why not send these arguments directly to the hook?

File: ui/app/src/components/ItemStatesManagement/ItemStatesManagement.tsx [hunk 2]
```diff
@@ -247,7 +247,7 @@ export default function ItemStatesManagement(props: ItemStatesManagementProps) {
         title={!embedded && <FormattedMessage id="siteTools.itemStates" defaultMessage="Item States" />}
         rightContent={
           <Button
-            className={!embedded && classes.filterButton}
+            className={!embedded ? classes.filterButton : ''}
```
Comment (rart) [inline ui/app/src/components/ItemStatesManagement/ItemStatesManagement.tsx]:
```suggestion
            className={embedded ? '' : classes.filterButton}
```
just for fun & clarity

File: ui/app/src/utils/hooks/useOnClose.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  return (onClose, reason, disableBackdropClick = false, disableEscapeKeyDown = false) => {
+    if (disableBackdropClick && reason === 'backdropClick') {
+      return false;
+    }
+
+    if (disableEscapeKeyDown && reason === 'escapeKeyDown') {
+      return false;
+    }
+
+    onClose?.();
```
Comment (rart) [inline ui/app/src/utils/hooks/useOnClose.ts]: Should propagate arguments received by the original event.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/utils/hooks/useOnClose.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+interface UseOnCloseProps {
+  onClose?(event: {}, reason: 'backdropClick' | 'escapeKeyDown'): void;
```
Comment (rart) [inline ui/app/src/utils/hooks/useOnClose.ts]: This isn't the right type. Even any is preferable here.
```suggestion
  onClose?(event: any, reason: 'backdropClick' | 'escapeKeyDown'): void;
```

**Review by rart — APPROVED** [review]

**Pull request merged**